leadership/README.md

```diff
@@ -57,3 +55,30 @@ for leader := <-leaderCh {
 
 A typical use case for this is to be able to always send requests to the current
 leader.
+
+## Fault tolerance
+
+Leadership returns an error channel for Candidates and Followers that you can use
```

Comment:
The initial intent of having an error chan is for the client to log errors instead of the Leadership package which is meant to be used as a library (which will potentially be in its own repository like `libkv`). Also `campaign` and `follow` are both running inside a goroutine in the library so you need a way to communicate back potential errors. Also the error chan is optional so the first version of the code still works.

Regarding the annoying usage of two `for` loops and a `label`, the only way I see to get rid of this is to do something like:

```go
func setupReplication(...) {
    [...]
    go run(candidate, server, primary, replica)
    go follow(follower, replica, addr)
    [...]
}

func run(candidate *leadership.Candidate, server *api.Server, primary *mux.Router, replica *api.Replica) {
    electedCh, errCh := candidate.RunForElection()
    for {
        select {
        case isElected := <-electedCh:
		if isElected {
			log.Info("Leader Election: Cluster leadership acquired")
			server.SetHandler(primary)
		} else {
			log.Info("Leader Election: Cluster leadership lost")
			server.SetHandler(replica)
		}

        case err := <-errCh:
            log.Error(err)
            time.Sleep(defaultRecoverTime)
            go run(candidate, server, primary, replica)
            return
        }
    }
}
```

This is *meh* to me, as we exit and create a new goroutine at each attempt. Is there any more elegant solution than this? Anyway we can still make the usage of Leadership more convenient later on.